### PR TITLE
Use the new lua UserInterface in the missions

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

File: mods/ra/maps/intervention/intervention.lua [hunk 1]
```diff
@@ -207,6 +228,10 @@ Tick = function()
 			player.MarkFailedObjective(villageObjective)
 		end
 	end
+
+	local percentage = 100 - VillageHousesLeft() * 10
+	local color = GetColor(percentage)
+	UserInterface.SetMissionText(percentage .. "% of the village destroyed.", color)
 end
```
Comment (obrakmann) [inline mods/ra/maps/intervention/intervention.lua]: You could move this into an `OnKilled` trigger set on every house in the village. Then this wouldn't need to run every tick.

Comment (abcdefg30) [inline mods/ra/maps/intervention/intervention.lua]: Done.

File: mods/ra/maps/survival01/survival01.lua [hunk 2]
```diff
@@ -345,6 +368,7 @@ InitMission = function()
 	end)
 
 	Trigger.AfterDelay(DateTime.Seconds(1), function() Media.PlaySpeechNotification(allies, "MissionTimerInitialised") end)
+	TimerColor = HSLColor.New(115, 240, 130)
```
Comment (Mailaender) [inline mods/ra/maps/survival01/survival01.lua]: `Player.Color` may be more convenient here. You just need to set it up with the Lua API in the `PlayerProperties.cs`.

**Comment by abcdefg30** [comment]
Updated.
This now also adds:
- Default color values (e.g. `HSLColor.White`)
- Default facings (e.g. `Facing.NorthEast`)
- More player properties:
  - color
  - race
  - spawnpoint ID
  - team ID
  - a IsBot bool

**Comment by abcdefg30** [comment]
By the way: The facing defaults are from our [old lua API](https://github.com/OpenRA/OpenRA/blob/6640fa2539cf37aca63553447a71d80266fc85b6/mods/common/lua/facing.lua).

**Comment by Mailaender** [comment]
Can you split off the facings into another pull request? They seem to be unused and unrelated here.

**Comment by abcdefg30** [comment]
Done: #8275

**Comment by Mailaender** [comment]
This is a well thought through enhancement of the Lua API. :+1: I could not find any regressions during testing. :white_check_mark:

**Comment by obrakmann** [comment]
:+1: Looks good to me

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/4d74ad9267844f9ec8fd63a0304591833a570cb4%5E...4d74ad9267844f9ec8fd63a0304591833a570cb4)

**Pull request merged**